sdgx/models/components/optimize/sdv_ctgan/data_transformer.py

```diff
@@ -79,15 +84,30 @@ def _fit_discrete(self, data):
                 A ``ColumnTransformInfo`` object.
         """
         column_name = data.columns[0]
-        ohe = OneHotEncoder()
-        ohe.fit(data, column_name)
-        num_categories = len(ohe.dummies)
+
+        encoder = OneHotEncoder()
+        encoder.fit(data, column_name)
+        num_categories = len(encoder.dummies)
+        activate_fn = "softmax"
+
+        checked = (
+            self.metadata.check_categorical_threshold(num_categories) if self.metadata else False
+        )
+        if encoder_type == "onehot" or not checked:
+            pass
+        elif encoder_type == "label":
```

Comment:
Have you ever try Frequency Encoder? I have wrote implement in `StatisticDataTransformer`. Most of my case, it seems that better quality than label encoding.

Reply:
Thanks for recommendation. I will add a FrequencyEncoder choice for it.